scalafix-core/shared/src/main/scala/scalafix/internal/rule/Disable.scala

```diff
@@ -1,18 +1,59 @@
 package scalafix.internal.rule
 
 import metaconfig.{Conf, Configured}
+
 import scala.meta._
+import scala.meta.transversers.Traverser
+import scalafix.config.CustomMessage
 import scalafix.internal.config.DisableConfig
-import scalafix.lint.LintCategory
-import scalafix.lint.LintMessage
-import scalafix.rule.SemanticRule
-import scalafix.rule.{Rule, RuleCtx}
-import scalafix.util.SemanticdbIndex
-import scalafix.util.SymbolMatcher
+import scalafix.internal.util.SymbolOps
+import scalafix.lint.{LintCategory, LintMessage}
+import scalafix.rule.{Rule, RuleCtx, SemanticRule}
+import scalafix.util.{SemanticdbIndex, SymbolMatcher}
+
+object Disable {
+
+  /**
+    * Searches for specific nodes in Tree and adds result of searching to the list.
+    * Also passes context through tree traversal.
+    * fn can either return a value. that stops the searching, or change the context.
```

Comment:
```
@param fn If fn returns Left then XXX. If fn returns Right then YYY.
```

It's not immediately clear from the signature what `fn` does.

Reply:
I know this PR is just moving code around but given that `Disable` is covering more use-cases after this PR we should try to document better what SearcherWithContext does.